src/test/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReaderTest.java

```diff
@@ -0,0 +1,147 @@
+                "PREFIX lr:     <http://www.example.org/se/liu/ida/hefquin/lpg2rdf#>\n"
+              + "PREFIX ex:     <http://example.org/>\n"
+              + "PREFIX xsd:     <http://www.w3.org/2001/XMLSchema#>\n"
+              + "\n"
+              + "ex:LPGtoRDFConfig\n"
+              + "   a  lr:LPGtoRDFConfiguration ;\n"
+              + "   lr:labelPredicate  \"http://www.w3.org/2000/01/rdf-schema#label\"^^xsd:anyURI ;\n"
+              + "   lr:nodeMapping  ex:IRINodeMapping ."
+              + "\n"
+              + "ex:IRINodeMapping\n"
+              + "   a  lr:IRIBasedNodeMapping ;\n"
+              + "   lr:prefixOfIRIs  \"https://example.org/node/\"^^xsd:anyURI .";
+
+        final Model lpg2rdf = ModelFactory.createDefaultModel();
+
+        final RDFParserBuilder b = RDFParser.fromString(turtle);
+        b.lang( Lang.TURTLE );
+        b.parse(lpg2rdf);
+
+        final LPG2RDFConfiguration lpg2RDFConfiguration = LPG2RDFConfigurationReader.readFromModel(lpg2rdf);
+        assert(lpg2RDFConfiguration.getLabel().isURI());
+        assert(lpg2RDFConfiguration.getLabel().getURI().equals("http://www.w3.org/2000/01/rdf-schema#label"));
+        final LPGNode node = new LPGNode("0", null, null);
+        final Node resultNode = lpg2RDFConfiguration.mapNode(node);
+        assertNotNull(resultNode);
+        assertTrue(resultNode.isURI());
+        assertEquals(resultNode.getURI(), "https://example.org/node/0");
+    }
+
+
+    @Test
+    public void LPG2RDFConfigWithBNodeBasedNodeMapping() {
+        final String turtle =
+                  "PREFIX lr:     <http://www.example.org/se/liu/ida/hefquin/lpg2rdf#>\n"
+                + "PREFIX ex:     <http://example.org/>\n"
+                + "PREFIX xsd:     <http://www.w3.org/2001/XMLSchema#>\n"
+                + "\n"
+                + "ex:LPGtoRDFConfig\n"
+                + "   a  lr:LPGtoRDFConfiguration ;\n"
+                + "   lr:labelPredicate  \"http://www.w3.org/2000/01/rdf-schema#label\"^^xsd:anyURI ;\n"
+                + "   lr:nodeMapping  ex:BNodeMapping ."
+                + "\n"
+                + "ex:BNodeMapping\n"
+                + "   a  lr:BNodeBasedNodeMapping .";
+
+        final Model lpg2rdf = ModelFactory.createDefaultModel();
+
+        final RDFParserBuilder b = RDFParser.fromString(turtle);
+        b.lang( Lang.TURTLE );
+        b.parse(lpg2rdf);
+
+        final LPG2RDFConfiguration lpg2RDFConfiguration = LPG2RDFConfigurationReader.readFromModel(lpg2rdf);
+        assert(lpg2RDFConfiguration.getLabel().isURI());
+        final LPGNode node = new LPGNode("0", "", null);
+        final Node resultNode = lpg2RDFConfiguration.mapNode(node);
+        assertNotNull(resultNode);
+        assertTrue(resultNode.isBlank());
+        assertEquals(resultNode.getBlankNodeId().toString(), "0");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void InvalidLabelLPG2RDFConfigWithIRIBasedNodeMapping() {
+        final String turtle =
```

Comment:
Can you please include a comment within this function to briefly describe what exactly is illegal in the given RDF description (i.e., to give other developers an idea of what exactly the test is checking).

Reply:
Same for the other unit tests below.